src/main/java/org/jenkinsci/test/acceptance/plugins/git/GitScm.java

```diff
@@ -190,9 +239,93 @@ public GitScm remoteName(String name) {
         return this;
     }
 
+    /**
+     * Select repository browser type
+     * @param name Type of repository browser
+     * @return this, to allow function chaining
+     */
+    public GitScm repositoryBrowser(String name) {
+        Select select = new Select(this.repositoryBrowser.resolve());
+        select.selectByVisibleText(name);
+        return this;
+    }
+
+    /**
+     * Set URL for repository browser
+     * @param url URL to be set
+     * @return this, to allow function chaining
+     */
+    public GitScm urlRepositoryBrowser(String url) {
+        this.urlRepositoryBrowser.set(url);
+        return this;
+    }
+
+    /**
+     * Set project name for gitblit and viewgit
+     * @param projectName value to be set
+     * @return this, to allow function chaining
+     */
+    public GitScm projectName(String projectName) {
+        this.projectName.set(projectName);
+        return this;
+    }
+
+    /**
+     * Set version for gitlab
+     * @param version value to be set
+     * @return this, to allow function chaining
+     */
+    public GitScm gitlabVersion(String version) {
+        this.gitlabVersion.set(version);
+        return this;
+    }
+
+    /**
+     * Set repository for phabricator
+     * @param repo value to be set
+     * @return this, to allow function chaining
+     */
+    public GitScm phabricatorRepo(String repo) {
+        this.phabricatorRepo.set(repo);
+        return this;
+    }
+
+    /**
+     * Add behaviour "Merge before build"
+     *
+     * @return behaviour, to access its method
+     */
+    public MergeBeforeBuild mergeBeforeBuild() {
+        return addBehaviour(MergeBeforeBuild.class);
+    }
+
+
     public <T extends Behaviour> T addBehaviour(Class<T> type) {
         control("hetero-list-add[extensions]").click();
-        return newInstance(type, this, "extensions");   // FIXME: find the last extension added
```

Comment:
Oh, the original logic here is quite tangled. `addBehaviour` opens the menu, creates the instance of page area (that does not yet exist!) and let that instance to do the menu selection. It would probably be better if we expose the selector/caption over `@Describable` annotation so `addBehaviour` knows how to select it so we can do things in proper order: open menu, select, instantiate page area. 

With that change done, we can benefit from `Control.selectDropdownMenu`* methods that already workarounds the problem with menu scrolling.

Reply:
Thanks for the suggestions. I didn't know this behaviour using `@Describable`.